### PR TITLE
fix:Alphabetical order in Trusted Merchants

### LINKED ISSUE #323: Alphabetical order is missing in Trusted Merchants
**Summary:**
.
Alphabetical order is missing in Trusted Merchants

**Device and Android version:**

Device Real Me 2
Android Version 8.1.0

**Screenshots:**

videotogif_2019 02 17_09 11 14

### PR DESCRIPTION
**Pull request by shaliniv16**
Fixes #323 

**Gif:**

videotogif_2019 02 25_21 26 06

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by shaliniv16** [comment]
@naman14 @luckyman20 please review.

**Comment by naman14** [comment]
@shaliniv16 https://github.com/openMF/mobile-wallet/pull/357#issuecomment-467962432
Will be merging this for now but next time please try to keep a single PR for a particular fix.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/presenter/MerchantsPresenter.java [hunk 1]
```diff
@@ -93,5 +95,12 @@ public void onFailure(String message) {
             }
         });
 
+        Collections.sort(savingsWithAssociationsList, new Comparator<SavingsWithAssociations>() {
```
Comment (naman14) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/presenter/MerchantsPresenter.java]: Why are you doing the sorting here. This should be done in `onComplete` before passing the data to the UI.

Comment (naman14) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/presenter/MerchantsPresenter.java]: In the current state, you are setting incorrect `externalIds` to each item in the list. The list  on which `taskLooper` was doing its work will be different from the list on which you will be setting externId `savingsWithAssociationsList.get(index).setExternalId( responseValue.getClient().getExternalId());`
cause the list has been sorted in between. 

Let the taskLooper set externalId first to all the list items and then in `onComplete`, sort the final list

Comment (shaliniv16) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/presenter/MerchantsPresenter.java]: @naman14 Updated according to the changes suggested.

**Pull request merged**